gems/aws-sdk-s3/lib/aws-sdk-s3/access_grants_credentials_provider.rb

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+module Aws
+  module S3
+    # @api private
+    ACCESS_GRANTS_CREDENTIALS_CACHE = LRUCache.new
```

Comment:
It might be better to memoize access to these so that we're not creating caches unless they are actually used.  I know they're not huge, but I think a general best practice would be to not create caches that aren't going to be used.
Also - We may want to specify the size here when creating them instead of relying on the default in core (letting us more easily change the default in core without impacting the value used for access grants in the future).

Reply:
I did this for all 3 caches. I don't think it's breaking but please take a look.